_pages/extension_sharing_strategies.md

```diff
@@ -0,0 +1,143 @@
+* **“Internal NDX”** refers to extensions used within a particular lab, organization or project that are not intended for use outside of the particular organization.
+* **“Public NDX”** refers to extensions intended for use by the public (or specific larger community).
+* **“nwb-extensions”** refers to the public catalog for NDX. The nwb-extensions catalog is implemented via a dedicated public GitHub organization at [https://github.com/nwb-extensions](https://github.com/nwb-extensions)
+* **“ndx-template”** repository refers to the code repository used to manage the sources of the extension
+   * **“ndx-custom>”** refers to a specific user extension repository that has usually been created from the ndx-template. This repository lives in the users own Git space and is NOT part of the nwb-extensions organization.
+* **“staged-extensions”** repository refers to the template Git repository as part of the NDX catalog used to register the extension with the catalog (this includes basic metadata about the NDX and information needed for install and deployment).
+    * **“ndx-custom-recipe”** refers to a specific recipe for deploying a particular extension. This repo is also part of the “nwb-extensions” GitHub organization. This repo should be automatically generated by the CI of the staged-extensions repo
+    **“user/ndx-custom-recipe”** refers to the users fork of the ndx-custom-recipe repo. This repo lives in the users Git space (NOT in nwb-extensions)
+
+### 1.2 Summary
+
+NWB:N defines a set of <a href="{{ site.url }}{{ site.baseurl }}/versioning_guidelines">Versioning Guidelines</a> for versioning extension namespaces and  <a href="{{ site.url }}{{ site.baseurl }}/ndx_sharing_guideline">NDX Sharing Guidelines</a> that public NDX MUST follow and internal NDXs are highly RECOMMENDED to follow. In order to help ensure compliance with these rules NWB:N provides a number of tools, templates, and guidelines.
+
+## 2. Methods and tools for sharing extensions
+
+### 2.1 [nwb-extensions](https://github.com/nwb-extensions): Extensions GitHub Organization
+
+NDX should be shared via a dedicated [nwb-extensions](https://github.com/nwb-extensions) GitHub organization. The NDX catalog provides the ndx-template and staged-extensions extension template repositories (see below) as well as all public ndx-custom-recipe repositories. In this way, all public extensions are managed in one place and MUST follow the same basic internal organization. This also avoids cluttering up the regular [Neurodata Without Borders GitHub organization](https://github.com/NeurodataWithoutBorders). Finally, this strategy allows for labs/universities/independent-groups to maintain the ndx-template source code repository in their own GitHub organization. The ability to do so, can be important in particular when the developers of the extension are funded by their own grants and/or are applying for funding.
```

Comment:
Do we really want this as a separate organization? It looks like there are 3 repositories here, the template, the stage, and the website. I feel like the website could easily be folded into neurodatawithoutborders.io, and should probably me managed there. I also think it wouldn't be that big of a deal to include these repos in the NWB organization (especially after we do a little spring cleaning over there)

Reply:
Do new NDXs get added to here as distinct repos? If so then I see why we need a separate one.

Reply:
Every user extension will have its own ``ndx-custom-recipe`` repo that is part of this organization. Mixing all the extension repos with the main NWB organization will get really messy and will make it hard to manage these repos as well to cleanly develop the infrastructure for the catalog. Similarly, as we built out the web interface for this, it will be useful to have a dedicated place for the technical development, as this quickly become more complicated. Also, from a program perspective it is helpful to have a distinct place for the catalog (both with regard to funders and for users). As such, I advocate for having the two separate organizations and websites. However, I do agree that the catalog website (once ready) must be accessible via the main neurodatawithoutborders.github.io website.